Task-52369: cannot upload a new version of a document when its name includes the character "+".

montassarhamadi opened the pull request:
Problem: Before this fix, it was impossible to upload a new version of the document in file explorer if the filename contains a +. when checking the existence of the file, the field currentFolder contains the filename, and the + is read as a space.

Fix: This change use function encodeURIComponent for variable current folder, so that, the + is transformed to %2B and then in the rest service, we decode it with URLDecoder.decode, and we replace the parameter string "UTF-8" by StandardCharsets.UTF_8 to avoid errors in code smells, so that we are able to see the +.

houssemexo26 committed:
TASK-35704: Create FB maintenance and update projects versions/dependencies

houssemexo26 committed:
TASK-35704: Create FB maintenance and update projects versions/dependencies

wassimmhamdi committed:
Task-48942: the comment for the versions is added to the activity of the initial document (copied/pasted) (#1577)

Problem: The comment of the version of the documentY is added in the activity of the documentX
Fix: Version comment of documentY are added to the correct activity of documentY

wassimmhamdi committed:
Task-48942: the comment for the versions is added to the activity of the initial document (copied/pasted) (#1577)

Problem: The comment of the version of the documentY is added in the activity of the documentX
Fix: Version comment of documentY are added to the correct activity of documentY

hakermi committed:
MAINT-36446: Display i18n drives labels in drives selector of the file attachment drawer (#1584)

ISSUE: The drives selector in the file attachment drawer was displaying the english default drives labels
FIX: Get the i18n drive label from the title and display it instead of the default one

hakermi committed:
MAINT-36446: Display i18n drives labels in drives selector of the file attachment drawer (#1584)

ISSUE: The drives selector in the file attachment drawer was displaying the english default drives labels
FIX: Get the i18n drive label from the title and display it instead of the default one

montassarhamadi committed:
Task-52369: cannot upload a new version of a document when its name includes the character "+".
Problem: Cannot upload a new version of a document which his name includes the character "+".
Fix: I should be able to upload  a new version even though the name of the document includes the character "+".

rdenarie committed:
Merge branch 'feature/maintenance' into task-52369

rdenarie commented on apps/portlet-explorer/src/main/webapp/javascript/eXo/ecm/MultiUpload.js:
```diff
@@ -283,7 +283,7 @@
 	
 		if (!oldName || oldName==null ||oldName==undefined) return oldName;
-		specialChar = "[]/'\":;";
+		specialChar = "[]'\":;";
```
As explained yesterday, you cannot remove a special character like this. It is a major risk to create side effects

montassarhamadi replied:
ok, Romain .I left it as is and used the function encodeURIComponent to encode the special characters in URI which be affected in param "currentFolder".

rdenarie reviewed (changes requested):
In addition, can you fix the sonar code smell ?
You can use this to replace the string UTF-8 : StandardCharsets.UTF_8

hakermi committed:
MAINT-52375: Avoid sql/injection in document service (#1587)

ISSUE: When calling rest service /portal/rest/documents/folder?folder=value, The user can forge his own query to acces other documents.
even As jcr sql is readonly, and that the returned data is checked with ACLs, there is no risk for data modification or exposure;but it is better to clean the variable folder before using it.
FIX: This PR should escape the injected string to prevent any dangerous chars that can cheat the sql interpreter

montassarhamadi committed:
used the function encodeURIComponent to encode the special characters in uri

montassarhamadi committed:
Merge branch 'task-52369' of github.com:exoplatform/ecms into task-52369

rdenarie committed:
Merge branch 'feature/maintenance' into task-52369

montassarhamadi committed:
replace string utf-8 by StandardCharsets.UTF_8

montassarhamadi committed:
Merge branch 'task-52369' of github.com:exoplatform/ecms into task-52369

rdenarie committed:
Merge branch 'feature/maintenance' into task-52369

rdenarie reviewed (changes requested):
Code is ok
But, can you update your PR description ? 

The first line should describe what is the problem, but you should add more context ?
Your first line should start by something like "Before this fix, it was impossible to upload a new version of the document in file explorer if the filename contains a +. when checking the existence of the file, the field currentFolder contains the filename, and the + is read as a space"

The second line should describe how you fix the problem : 
"This change use function encodeURIComponent for variable current folder, so that, the + is transformed to %2B and then in the rest service, we decode it with URLDecoder.decode, so that we are able to see the +"
